sdk/python/kfp/dsl/for_loop.py

```diff
@@ -180,7 +199,7 @@ def from_raw_items(
         return LoopArgument(
             items=raw_items,
             name_code=name_code,
-            channel_type=type(raw_items[0]).__name__,
+            channel_type=_get_first_element_type(raw_items),
```

Comment:
Can you add an E2E compilation test to [this directory](https://github.com/kubeflow/pipelines/tree/master/sdk/python/test_data/pipelines) and [this config](https://github.com/kubeflow/pipelines/blob/5b7f67acdcbd81d612a3deb39823f28ac6a56c6e/sdk/python/test_data/test_data_config.yaml#L19) to demonstrate that this solves the specific `dsl.ParallelFor` issue?

To see how these files/config are used, take a look at the test entrypoint [here](https://github.com/kubeflow/pipelines/blob/5b7f67acdcbd81d612a3deb39823f28ac6a56c6e/sdk/python/kfp/compiler/read_write_test.py#L166).